doc/sphinx/proofs/writing-proofs/proof-mode.rst

```diff
@@ -1,74 +1,173 @@
 .. _proofhandling:
 
--------------------
- Proof handling
--------------------
+----------
+Proof mode
+----------
 
-In Coq’s proof editing mode all top-level commands documented in
-Chapter :ref:`vernacularcommands` remain available and the user has access to specialized
-commands dealing with proof development pragmas documented in this
-section. They can also use some other specialized commands called
-*tactics*. They are the very tools allowing the user to deal with
-logical reasoning. They are documented in Chapter :ref:`tactics`.
+:gdef:`Proof mode <proof mode>` is used to prove theorems.
+Coq enters proof mode when you begin a proof,
+such as with the :cmd:`Theorem` command.  It exits proof mode when
+you complete a proof, such as with the :cmd:`Qed` command.  Tactics,
+which are available only in proof mode, incrementally transform incomplete
+proofs to eventually generate a complete proof.
 
-Coq user interfaces usually have a way of marking whether the user has
-switched to proof editing mode. For instance, in coqtop the prompt ``Coq <``   is changed into
-:n:`@ident <`   where :token:`ident` is the declared name of the theorem currently edited.
+When you run Coq interactively, such as through CoqIDE, Proof General or
+coqtop, Coq shows the current proof state (the incomplete proof) as you
+enter tactics.  This information isn't shown when you run Coq in batch
+mode with `coqc`.
 
-At each stage of a proof development, one has a list of goals to
-prove. Initially, the list consists only in the theorem itself. After
-having applied some tactics, the list of goals contains the subgoals
-generated by the tactics.
+Proof State
+-----------
 
-To each subgoal is associated a number of hypotheses called the *local context*
-of the goal. Initially, the local context contains the local variables and
-hypotheses of the current section (see Section :ref:`gallina-assumptions`) and
-the local variables and hypotheses of the theorem statement. It is enriched by
-the use of certain tactics (see e.g. :tacn:`intro`).
+The :gdef:`proof state` consists of one
+or more unproven goals.
+Each goal has a :gdef:`conclusion` (the statement that is to be proven)
+and may have zero or more named facts that can be used in proving the
+conclusion.  The proof may also use facts from the :term:`global environment`
+such as definitions and proven theorems.
```

Comment:
If the definition of "local context" is now elsewhere, this could be reformulated to something simpler like:

```suggestion
Each goal has a :gdef:`conclusion` (the statement that is to be proven)
and a :term:`local context`, which contains local :term:`variables <variable>` that can
be used in proving the conclusion.  The proof may also use :term:`constants <constant>`
from the :term:`global environment` such as definitions and proven theorems.
```

Reply:
Check what I wrote.  We don't have a glossary entry for `constant` or `variable` yet.  Not sure it's necessary or useful to define them since they are such pervasive concepts in math and CS.

Reply:
It can wait. But variable is a pervasive concept both in math and CS indeed but with slightly differing meanings (CS variables being often mutable).